service/history/workflow/mutable_state_impl.go

```diff
@@ -2650,6 +2649,44 @@ func (ms *MutableStateImpl) ApplyWorkflowExecutionStartedEvent(
 			//nolint:staticcheck // SA1019: worker versioning v0.31
 			ms.executionInfo.VersioningInfo.VersioningOverride.Behavior = enumspb.VERSIONING_BEHAVIOR_UNSPECIFIED
 		}
+
+		if ms.executionInfo.GetVersioningInfo().GetVersioningOverride().GetPinned() != nil {
+			if ms.HasParentExecution() && ms.GetExecutionInfo().GetNamespaceId() != ms.GetExecutionInfo().GetParentNamespaceId() {
+				// don't inherit pinned version if child is in a different namespace
+				ms.executionInfo.VersioningInfo.VersioningOverride = nil
+			}
+			if event.GetPreviousRunVersioningInfo().GetTaskQueue() != "" ||
+				event.GetParentVersioningInfo().GetTaskQueue() != "" {
+				// TODO(carlydf): Check if new wf Task Queue belongs to the override version
+				// for now, this means no inheritance if pinned override
```

Comment:
if both tq names are present and different from each other, then we don't know which one we should check for belonging in the version. we don't know where the override came from.